dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -123,42 +122,42 @@ internal async Task<string> RenderAsync(List<Block> blocks, Kernel kernel, Kerne
         return resultString;
     }
 
-    /// <summary>
-    /// Given a list of blocks, render the Variable Blocks, replacing placeholders with the actual value in memory.
-    /// </summary>
-    /// <param name="blocks">List of blocks, typically all the blocks found in a template.</param>
-    /// <param name="arguments">Arguments to use for rendering.</param>
-    /// <returns>An updated list of blocks where Variable Blocks have rendered to Text Blocks.</returns>
-    internal IList<Block> RenderVariables(IList<Block> blocks, KernelArguments? arguments)
+    private static string? ConvertToString(object? value, CultureInfo culture)
     {
-        this._logger.LogTrace("Rendering variables");
-        return blocks.Select(block => block.Type != BlockTypes.Variable
-            ? block
-            : new TextBlock(((ITextRendering)block).Render(arguments), this._loggerFactory)).ToList();
+        if (value == null) { return null; }
+
+        var sourceType = value.GetType();
+
+        var converterFunction = GetTypeConverterDelegate(sourceType);
+
+        return converterFunction == null
+            ? value.ToString()
+            : converterFunction(value, culture);
     }
 
-    /// <summary>
-    /// Validates that all the KernelArguments are of string type.
-    /// </summary>
-    /// <param name="arguments">The collection of KernelArguments to validate.</param>
-    /// <exception cref="KernelException">Thrown when an argument is not of type string.</exception>
-    private static void AssertArgumentOfStringType(KernelArguments? arguments)
-    {
-        if (arguments == null)
+    private static Func<object?, CultureInfo, string?>? GetTypeConverterDelegate(Type sourceType) =>
```

Comment:
This is the duplicated logic you talk about cleaning up subsequently?

Reply:
Right, both `ConvertToString` and `GetTypeConverterDelegate` method will be moved to a separate class and reused by the `KErnelPromptTemplate` and `FunctionResult` classes.